app/controllers/saasbook_controller.rb

```diff
@@ -27,6 +27,10 @@ def preface
   def show_section
     @chapter_id = params[:chapter_id]
     @section_id = params[:section_id]
+    if Integer(@section_id) <= 0
```

Comment:
This has nothing to do with annotations; this is for redirecting malformed URLs to a correct place. For example, `https://esaas169l.herokuapp.com/chapter/1/section/0` would now redirect to `https://esaas169l.herokuapp.com/chapter/1` instead of throwing an error.

Take a look at the other controller actions in this file to see how and when the -1 flag is relevant.